src/dune_rpc_server/dune_rpc_server.ml

```diff
@@ -93,6 +93,7 @@ module Session = struct
     type 'a t =
       { queries : Packet.t Fiber.Stream.In.t
       ; id : Id.t
+      ; mutable menu : Menu.t option
```

Comment:
Isn't it better to expose such fields (that are set only once) via an `Ivar.t` instead, so one can reliably wait for them to be filled?

Reply:
When it's necessary to wait for them, then sure. But our case is for `dune rpc status`. Which prints all the connected clients and the menus they're connected with. So we would be using `Ivar.peek` anyway, because we want to return the status immediately rather than waiting for clients to version negotiate. So if we turn it into an Ivar, we're not really going to be using the reliable waiting.